[Profiler] Move Allocation into EventType.

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #77697
* #77696
* #77695
* __->__ #77694

Continuing the trend of unification, this PR removes the special path for allocation tracking. The overall delta is pretty minimal; it's mostly just extending and unifying visitors. This also comes with the added benefit that memory profiling now gets to take advantage of the lock-free machinery.

Differential Revision: [D36189043](https://our.internmc.facebook.com/intern/diff/D36189043/)

Commit:
[Profiler] Move Allocation into EventType.

Continuing the trend of unification, this PR removes the special path for allocation tracking. The overall delta is pretty minimal; it's mostly just extending and unifying visitors. This also comes with the added benefit that memory profiling now gets to take advantage of the lock-free machinery.

Differential Revision: [D36189043](https://our.internmc.facebook.com/intern/diff/D36189043/)

[ghstack-poisoned]

aaronenyeshi reviewed (approved):
Reviewed internally, LGTM!

Commit:
Update on "[Profiler] Move Allocation into EventType."


Continuing the trend of unification, this PR removes the special path for allocation tracking. The overall delta is pretty minimal; it's mostly just extending and unifying visitors. This also comes with the added benefit that memory profiling now gets to take advantage of the lock-free machinery.

Differential Revision: [D36189043](https://our.internmc.facebook.com/intern/diff/D36189043/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Move Allocation into EventType."


Continuing the trend of unification, this PR removes the special path for allocation tracking. The overall delta is pretty minimal; it's mostly just extending and unifying visitors. This also comes with the added benefit that memory profiling now gets to take advantage of the lock-free machinery.

Differential Revision: [D36189043](https://our.internmc.facebook.com/intern/diff/D36189043/)

[ghstack-poisoned]
